src/v/raft/consensus.cc

```diff
@@ -3918,19 +3968,15 @@ void consensus::upsert_recovery_state(
     }
 }
 
-ss::future<> consensus::maybe_flush_log() {
-    // if there is nothing to do exit without grabbing an op_lock, this check is
-    // sloppy as we data can be in flight but it is ok since next check will
-    // detect it and flush log.
-    if (
-      _pending_flush_bytes < _max_pending_flush_bytes
-      && time_since_last_flush() < flush_ms()) {
-        co_return;
-    }
+ss::future<> consensus::do_flush() {
     try {
         auto holder = _bg.hold();
-        auto u = co_await _op_lock.get_units();
-        auto flushed = co_await flush_log();
+        auto flush = ssx::now(consensus::flushed::no);
+        {
+            auto u = co_await _op_lock.get_units();
```

Comment:
Can't this be moved inside of flush_log and skip the `co_await _log->flush();` call? I don't think that one needs the lock and is the one taking the most time.

Reply:
> Can't this be moved inside of flush_log and skip the co_await _log->flush(); call? I don't think that one needs the lock and is the one taking the most time.

I thought so too but I was a bit scared to touch this code 1 week before the release, perhaps we can do a follow up _after_ the release and unify all flush callers.